src/components/vm/nics/vmNicsCard.tsx

```diff
@@ -523,18 +523,24 @@ export class VmNetworkTab extends React.Component<VmNetworkTabProps, VmNetworkTa
         const id = vmId(vm.name);
 
         const onChangeState = (network: VMInterface) => {
-            return (e: React.MouseEvent) => {
+            return async (e: React.MouseEvent) => {
                 e.stopPropagation();
                 if (network.mac) {
-                    domainChangeInterfaceSettings({ vmName: vm.name, connectionName: vm.connectionName, macAddress: network.mac, state: network.state === 'up' ? 'down' : 'up', hotplug: vm.state === "running" })
-                            .then(() => domainGet({ connectionName: vm.connectionName, id: vm.id }))
-                            .catch(ex => {
-                                onAddErrorNotification({
-                                    text: cockpit.format(_("NIC $0 of VM $1 failed to change state"), network.mac, vm.name),
-                                    detail: ex.message,
-                                    resourceId: vm.id,
-                                });
-                            });
+                    try {
+                        await virtXmlEdit(
+                            vm,
+                            "network",
+                            { mac: network.mac },
+                            { link: { state: network.state === 'up' ? 'down' : 'up' } },
+                            { update: vm.state === "running" });
+                        domainGet({ connectionName: vm.connectionName, id: vm.id });
+                    } catch (ex) {
+                        onAddErrorNotification({
+                            text: cockpit.format(_("NIC $0 of VM $1 failed to change state"), network.mac, vm.name),
+                            detail: String(ex),
+                            resourceId: vm.id,
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2334-68be91e9-20250929-092324-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)